samples/drivers/counter/alarm/src/main.c

```diff
@@ -0,0 +1,66 @@
+/*
+ * Copyright (c) 2019 Linaro Limited
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <zephyr.h>
+
+#include <device.h>
+#include <counter.h>
+#include <misc/printk.h>
+
+#define DELAY 2000000
+#define ALARM_CHANNEL_ID 0
+
+struct counter_alarm_cfg alarm_cfg;
+
+static void test_counter_interrupt_fn(struct device *counter_dev,
+				u8_t chan_id, u32_t ticks, void *user_data)
+{
+	u32_t now_ticks = counter_read(counter_dev);
+	u64_t now_usec = counter_ticks_to_us(counter_dev, now_ticks);
+	int now_sec = (int)(now_usec / USEC_PER_SEC);
+	struct counter_alarm_cfg *config = user_data;
+
+	printk("!!! Alarm !!!\n");
+	printk("Now: %d\n", now_sec);
+
+	counter_disable_channel_alarm(counter_dev, ALARM_CHANNEL_ID);
```

Comment:
you shouldn't need to call that as alarms are single shot (i will prepare PR soon to clarify that in counter.h)